trip_app/trips/views.py

```diff
@@ -34,24 +38,40 @@ def get_queryset(self):
 
         time1 = query_params.get("time1")
         time2 = query_params.get("time2")
-        start_point = query_params.get("sp")
-        dest_point = query_params.get("dp")
 
         if time1 and time2:
             queryset = queryset.filter(dep_time__gt=time1, dep_time__lt=time2)
 
-        if start_point and dest_point:
-            start_point = str_to_geopoint(start_point)
-            dest_point = str_to_geopoint(dest_point)
-            # Annotate queryset with 2 attributes:
-            # dist1 - distance between user and trip start point;
-            # dist2 - between user and trip end point.
-            # Order the queryset in the ascending order by sum distance,
-            queryset = (
-                queryset.annotate(dist1=Distance("start_point", start_point))
-                .annotate(dist2=Distance("dest_point", dest_point))
-                .order_by(F("dist1") + F("dist2"))
-            )
+        point1 = query_params.get("point1")
+        point2 = query_params.get("point2")
+        address1 = query_params.get("addr1")
+        address2 = query_params.get("addr2")
+        geo_coords = bool(point1 and point2)
+        address_coords = bool(address1 and address2)
+
+        if geo_coords or address_coords:
+            try:
+                if geo_coords:
+                    geo_point1 = str_to_geopoint_or_validation_error(point1)
+                    geo_point2 = str_to_geopoint_or_validation_error(point2)
+                elif address_coords:
+                    geo_point1 = geocode_or_validation_error(address1)
+                    geo_point2 = geocode_or_validation_error(address2)
+            except ValidationError:
+                # return empty queryset
+                return Trip.objects.none()
```

Comment:
This case feels more like a `400` error to me, rather than `200` with empty list. Any specific reasoning on your choice?

Reply:
Nothing specific) . I just didn't know that DRF is smart enough to interpret ValidationError as 400 BadResponse, and didn't know how to return the proper response from here ) It look like I haven't to catch the exception anywore, wow. Thanks!

Reply:
This should be updated now.